app/scripts/modules/core/pipeline/config/preconditions/selector/preconditionSelector.directive.js

```diff
@@ -60,6 +60,16 @@ module.exports = angular.module('spinnaker.core.pipeline.config.preconditions.se
 
       let accountFilter = (cluster) => cluster.account === $scope.precondition.context.credentials;
       $scope.regions = appListExtractorService.getRegions([$scope.application], accountFilter);
+
+      //Setting cloudProvider when account is updated
+      let providerFilter = (accounts) => accounts.name === $scope.precondition.context.credentials;
```

Comment:
Very minor, but you're iterating over an array of `accounts`, and this filter will be called with an `account` (rather than all `accounts`).